### PR TITLE
Fix workbench crashing issue when save_as a running script (2nd try)

### PR DESCRIPTION
**Pull request by KedoKudo**
**Description of work.**

- Original Gitlab defect: [SCD435](https://code.ornl.gov/sns-hfir-scse/diffraction/single-crystal/single-crystal-diffraction/-/issues/435)
- This is #33081 into `main`

This PR fixes an issue where the workbench crashes when `save_as` a running script.
The issue behind is that original `save_as` closes the current running tab, resulting in an error when child thread is trying to report the completion of the script running.
This PR removes the force close of the running tab.

**To test:**

- Run the following script in workbench script editor.
```python
from time import sleep

sleep(60)
```
- Save it if it is not saved already
- While the script is running, use the `save_as` to save a copy of the script
- Instead of losing the current running tab, you should see a new tab created for the copy.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by zjmorgan** [comment]
The fix resolves the problem nicely. Here is an example intensive loading script that no longer crashes.

Documentation iand code changes are good. Approved.

```python
# import mantid algorithms, numpy and matplotlib
from mantid.simpleapi import *
import matplotlib.pyplot as plt
import numpy as np

import time

for i in range(10):
    ws = LoadEmptyInstrument(InstrumentName='CORELLI')

time.sleep(10)

for i in range(10):
    ws2 = LoadEmptyInstrument(InstrumentName='CORELLI')
    
print(mtd['ws2'].getInstrument())
```

**Review by zjmorgan — DISMISSED** [review]
All changes work well. Minimal solution works.

File: docs/source/release/v6.3.0/mantidworkbench.rst [hunk 1]
```diff
@@ -39,8 +40,11 @@ Bugfixes
 - Fixed a bug where the z-axis editor dialog was being initialised from the y-axis for a 3D plot.
 - Fixed a bug with autoscaling of colorfill plots from within the figure options.
 - Fixed an issue to plot negative values with logarithm scaling in slice view.
+<<<<<<< HEAD
```
Comment (peterfpeterson) [inline docs/source/release/v6.3.0/mantidworkbench.rst:43]: Remove the merge conflict markers

File: docs/source/release/v6.3.0/mantidworkbench.rst [hunk 2]
```diff
@@ -39,8 +40,11 @@ Bugfixes
...
 - Workbench will no longer hang if an algorithm was running when workbench was closed.
 - Stopped workbench from ignoring GUIs that want to cancel closing
 - Fixed a bug in the editor where uncommenting using 'ctrl+/' wasn't working correctly for lines of the form '<optional whitespace>#code_here # inline comment'.
+=======
```
Comment (peterfpeterson) [inline docs/source/release/v6.3.0/mantidworkbench.rst:47]: These too

**Review by peterfpeterson — CHANGES_REQUESTED** [review]

**Review by peterfpeterson — APPROVED** [review]

**Pull request merged**